hivemind_etl/storage/s3_client.py

```
@@ -0,0 +1,124 @@
+        )
+        return key
+
+    def get_data_by_key(self, key: str) -> Dict[str, Any]:
+        """Get data from S3 using a specific key."""
+        try:
+            obj = self.s3_client.get_object(Bucket=self.bucket_name, Key=key)
+            return json.loads(obj["Body"].read().decode("utf-8"))
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchKey":
+                logging.error(f"No data found for key: {key}")
+                raise ValueError(f"No data found for key: {key}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)